### PR TITLE
Createonlyfields setting

### PR DESCRIPTION
**Pull request by Jyggafey**
Introduces a setting containing field names that should not be used when updating a contact, removing the need for an argument with that purpose to api.sync_contacts.
The motivation for this was that emarsys does not properly protect the Opt-In field of contacts from changes through the api.

Also prepares the removal of the user argument to EventManager.trigger while fixing an edge case where the user given to the method is not the intended recipient of the email.

### CODE REVIEW

File: django_emarsys/api.py [hunk 1]
```diff
@@ -134,6 +133,7 @@ def sync_contacts(contacts, create_missing=True, create_only_fields=None,
 
     The dictionary keys are mapped to emarsys field ids using
     settings.EMARSYS_FIELDS, which can be generated with `get_fields()`.
+    Fields in settings.EMARSYS_CREATE_ONLY_FIELDS are not send when updating a contact.
```
Comment (mbertheau) [inline django_emarsys/api.py]: "are not sent"

Comment (mbertheau) [inline django_emarsys/api.py]: Please break the line before 80 characters so that `flake8 .` doesn't complain.

**Comment by mbertheau** [comment]
d05b0ba should be squashed into 8ee15db and the commit message adapted correspondingly.

LGTM otherwise. Thanks!

**Comment by Jyggafey** [comment]
commits cherry picked onto master

**Pull request closed without merging**